sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -614,14 +613,9 @@ object RocksDBStateStoreProvider {
 
   private val MAX_AVRO_ENCODERS_IN_CACHE = 1000
   // Add the cache at companion object level so it persists across provider instances
-  private val avroEncoderMap: NonFateSharingCache[String, AvroEncoder] = {
-    val guavaCache = CacheBuilder.newBuilder()
-      .maximumSize(MAX_AVRO_ENCODERS_IN_CACHE)  // Adjust size based on your needs
-      .expireAfterAccess(1, TimeUnit.HOURS)  // Optional: Add expiration if needed
-      .build[String, AvroEncoder]()
-
-    new NonFateSharingCache(guavaCache)
```

Comment:
Actually, I suspect the root cause is similar to SPARK-44064.

Reply:
In more detail, this is because during the Maven test process, the `sql/core` module uses the shaded `core` module. The shaded `core` module has relocation `com.google.common` to `org.sparkproject.guava`. 

As a result, the `com.google.common.cache.Cache` in the `NonFateSharingCache` constructor has been relocation to `org.sparkproject.guava.cache.Cache`, which leads to Maven test error in the current scenario.

So, if we add a new apply function unrelated to Guava in `NonFateSharingCache`, such as

```scala
def apply[K, V](
    maximumSize: Long,
    expireAfterAccessTime: Long,
    expireAfterAccessTimeUnit: TimeUnit): NonFateSharingCache[K, V] = {
    val builder = CacheBuilder.newBuilder().asInstanceOf[CacheBuilder[K, V]]
    if (maximumSize > 0L) {
      builder.maximumSize(maximumSize)
    }
    if(expireAfterAccessTime > 0) {
      builder.expireAfterAccess(expireAfterAccessTime, expireAfterAccessTimeUnit)
    }
    new NonFateSharingCache(builder.build[K, V]())
  }
``` 

and use the newly defined apply function to construct `NonFateSharingCache`, like `NonFateSharingCache(MAX_AVRO_ENCODERS_IN_CACHE, 1, TimeUnit.HOURS)` it should be able to avoid this testing issue.

Reply:
Thanks @LuciferYang